scio-core/src/main/scala/com/spotify/scio/io/Tap.scala

```diff
@@ -77,11 +79,12 @@ case class BigQueryTap(table: TableReference) extends Tap[TableRow] {
 }
 
 /** Tap for object files on local file system or GCS. */
-case class ObjectFileTap[T: ClassTag](path: String) extends Tap[T] {
+case class ObjectFileTap[T: ClassTag](path: String, coder: Coder[T] = null) extends Tap[T] {
   override def value: Iterator[T] = {
-    val coder = KryoAtomicCoder[T]
+    import com.spotify.scio.Implicits.RichCoderRegistry
```

Comment:
On second thought, add a `ScioUtil.getScalaCoder[T]` method and create new `CoderRegistry` plus implicits inside?